gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1308 @@
+        ttdas_sent = 0
+
+        for i in range(workers):
+            try:
+                parent_conn, child_conn = Pipe()
+
+                # Load Balancing, for example if there are 9 ttdas and 4 workers, the load will be balanced 2, 2, 2, 3.
+                n_ttdas = 0
+                if i == workers - 1:  # i is a index, hence -1
+                    # is this the last worker that needs to be created?
+                    # then task that worker with all the remaining models
+                    n_ttdas = len(self.ttda) - ttdas_sent
+                else:
+                    n_ttdas = int((len(self.ttda) - ttdas_sent) / (workers - i))
+
+                process = Process(
+                    target=self._asymmetric_distance_matrix_worker,
+                    args=(i, ttdas_sent, n_ttdas, child_conn, threshold, method),
+                )
+                ttdas_sent += n_ttdas
+
+                processes.append(process)
+                pipes.append((parent_conn, child_conn))
+                process.start()
+
+            except ProcessError:
+                self._teardown(pipes, processes, i)
+                raise ProcessError
+
+        distances = []
+        # note, that the following loop maintains order in how the ttda will be concatenated
+        # which is very important. Ordering in ttda has to be the same as when using only one process
+        for parent_conn, _ in pipes:
+            answer = parent_conn.recv()
+            parent_conn.close()  # child conn will be closed from inside the worker
+            # this does basically the same as the _generate_topic_models function (concatenate all the ttdas):
+            distances.append(answer[1])
+
+        # end all processes
+        for process in processes:
+            process.terminate()
+
+        self.asymmetric_distance_matrix = np.concatenate(distances)
+
+        return self.asymmetric_distance_matrix
+
+    def _calculate_asymmetric_distance_matrix_chunk(self, ttda1, ttda2, threshold, start_index, method):
+        """Calculate an (asymmetric) distance from each topic in ``ttda1`` to each topic in ``ttda2``.
+
+        Parameters
+        ----------
+        ttda1 and ttda2: 2D arrays of floats
+            Two ttda matrices that are going to be used for distance calculation. Each row in ttda corresponds to one
+            topic. Each cell in the resulting matrix corresponds to the distance between a topic pair.
+        threshold : float, optional
+            threshold defaults to: ``{"mass": 0.95, "rank": 0.11}``, depending on the selected method
+        start_index : int
+            this function might be used in multiprocessing, so start_index has to be set as ttda1 is a chunk of the
+            complete ttda in that case. start_index would be 0 if ``ttda1 == self.ttda``. When self.ttda is split into
+            two pieces, each 100 ttdas long, then start_index should be be 100. default is 0
+        method : {'mass', 'rank}, optional
```

Comment:
Is there a benefit to passing in a string and then doing the dictionary lookup instead of just passing the function object itself, directly?